### ISSUE TITLE
on_batch_end fails on last batch in epoch when model has multiple heads

### ISSUE THREAD

**Issue by tik0** [keras-team/keras#12005]
- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Jupyter Notebook with corresponding error in the last cell:** https://gist.github.com/tik0/0058322b966bfecfcc7a9ccdbeba3400/cd46df002d06111472d239b16f4c756d9ea93e0f

**Issue description:**
I've implemented the Keras VAE example (as explained in my SO question [here](https://stackoverflow.com/questions/54069363/output-multiple-losses-added-by-add-loss-in-keras))  with multiple loss-layer.
However, the function _on_batch_end_ fails on the last batch in epoch due to the fact that the last batch does not meet the _batch_size_. Let's consider training on MNIST with 60000 samples. With the batch_size of 128, the last batch has 60000%128=96 samples.
_v * batch_size_ has the size of 96 but _self.totals[k]_ still has the size 128. When I apply a batch_size of 100 (60000%100=0) everthing works as well as in the single headed example.

**Comment by tik0** [comment]
So the tensor given to add_loss is evaluated over the batch size. But the callback `on_batch_end` assumes a single value. Adding `K.mean()` to every loss fixes the issue for me as updated in the [gist](https://gist.github.com/tik0/0058322b966bfecfcc7a9ccdbeba3400/afbc71b3cbaaf7d5ae23382b74bf60390184f3d1). But the question remains, why this error occurs at the end of the epoch and does my approach only masks an issue?
